images/egress/router/egress-router.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# OpenShift egress router setup script
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+BLANK_LINE_OR_COMMENT_REGEX="([[:space:]]*$|#.*)"
+PORT_REGEX="[[:digit:]]+"
+PROTO_REGEX="(tcp|udp)"
+IP_REGEX="[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+"
+
+if [[ "${EGRESS_SOURCE:-}" !~ ^${IP_REGEX}$ ]]; then
+    echo "EGRESS_SOURCE unspecified or invalid"
+    exit 1
+fi
+if [[ -z "${EGRESS_DESTINATION:-}" ]]; then
+    echo "EGRESS_DESTINATION unspecified"
+    exit 1
+fi
+if [[ "${EGRESS_GATEWAY:-}" !~ ^${IP_REGEX}$ ]]; then
+    echo "EGRESS_GATEWAY unspecified or invalid"
+    exit 1
+fi
+
+set -x
+
+# The pod may die and get restarted; only try to add the
+# address/route/rules if they are not already there.
+if ! ip route get "${EGRESS_GATEWAY}" | grep -q macvlan0; then
+    ip addr add "${EGRESS_SOURCE}"/32 dev macvlan0
+    ip link set up dev macvlan0
+
+    ip route add "${EGRESS_GATEWAY}"/32 dev macvlan0
+    ip route del default
+    ip route add default via "${EGRESS_GATEWAY}" dev macvlan0
+fi
+
+# Update neighbor ARP caches in case another node previously had the IP. (This is
+# the same code ifup uses.)
+arping -q -A -c 1 -I macvlan0 "${EGRESS_SOURCE}"
+( sleep 2;
+  arping -q -U -c 1 -I macvlan0 "${EGRESS_SOURCE}" || true ) > /dev/null 2>&1 < /dev/null &
```

Comment:
Pretty sure I just copied the "|| true" from ifup-eth, although the current version doesn't have that... I'll look into why arping might fail and reconsider if want it